modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -834,7 +839,7 @@ public CompletableFuture<Ignite> start(Path configPath) {
 
                         // Start all other components after the join request has completed and the node has been validated.
                         try {
-                            lifecycleManager.startComponents(
+                            startFuturesAfterJoin.add(lifecycleManager.startComponents(
```

Comment:
Can we change `runAsync` to `applyAsync` here and return `startFuturesAfterJoin` as the result of this function?

Reply:
refactored